linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/CoreJSLib.scala

```diff
@@ -1494,34 +1491,30 @@ private[emitter] object CoreJSLib {
         val arrayClass = varRef("arrayClass")
         val typedArrayClass = varRef("typedArrayClass")
         val isAssignableFromFun = varRef("isAssignableFromFun")
+        val self = varRef("self")
         val that = varRef("that")
         val obj = varRef("obj")
         val array = varRef("array")
         MethodDef(static = false, Ident("initSpecializedArray"),
             paramList(componentData, arrayClass, typedArrayClass, isAssignableFromFun), {
           Block(
               initArrayCommonBody(arrayClass, componentData, componentData, 1),
-              privateFieldSet("isAssignableFromFun", isAssignableFromFun || {
-                Function(arrow = false, paramList(that), {
-                  Return(This() === that)
-                })
-              }),
+              const(self, This()), // capture `this` for use in arrow fun
+              privateFieldSet("isAssignableFromFun", isAssignableFromFun ||
+                  genArrowFunction(paramList(that), Return(self === that))),
```

Comment:
An argument that spans two lines should use its own lines:
```suggestion
              privateFieldSet("isAssignableFromFun",
                  isAssignableFromFun ||
                  genArrowFunction(paramList(that), Return(self === that))),
```
Alternatively, keep the `{}`:
```suggestion
              privateFieldSet("isAssignableFromFun", isAssignableFromFun || {
                genArrowFunction(paramList(that), Return(self === that))
              },
```